File: app/src/main/java/com/hjc/reader/ui/gank/adapter/WelfareAdapter.java
package com.hjc.reader.ui.gank.adapter;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.hjc.reader.R;
import com.hjc.reader.bean.response.GankDayBean;
import com.hjc.reader.databinding.ItemWelfareBinding;


public class WelfareAdapter extends BaseQuickAdapter<GankDayBean, BaseViewHolder> {

    public WelfareAdapter() {
        super(R.layout.item_welfare);
    }

    @Override
    protected void onItemViewHolderCreated(@NonNull BaseViewHolder viewHolder, int viewType) {
        DataBindingUtil.bind(viewHolder.itemView);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, GankDayBean item) {
        if (item == null) {
            return;
        }

        ItemWelfareBinding binding = helper.getBinding();
        if (binding != null) {
            binding.setGankDayBean(item);
        }

//        瀑布流布局
//        int position = helper.getAdapterPosition();
//        if (position % 2 == 0) {
//            ivPic.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, DensityUtil.dp2px(250f)));
//        } else {
//            ivPic.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,  DensityUtil.dp2px(300f)));
//        }
    }
}

File: module-gank/src/main/java/com/hjc/learn/gank/adapter/WelfareAdapter.kt
package com.hjc.learn.gank.adapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.hjc.learn.gank.R
import com.hjc.learn.gank.databinding.GankItemWelfareBinding
import com.hjc.library_net.model.GankDayBean

/**
 * @Author: HJC
 * @Date: 2021/2/23 17:13
 * @Description: 福利社区adapter
 */
class WelfareAdapter(data: MutableList<GankDayBean>?) :
    BaseQuickAdapter<GankDayBean, BaseViewHolder>(R.layout.gank_item_welfare, data) {

    override fun onItemViewHolderCreated(viewHolder: BaseViewHolder, viewType: Int) {
        DataBindingUtil.bind<ViewDataBinding>(viewHolder.itemView)
    }

    override fun convert(holder: BaseViewHolder, item: GankDayBean) {
        val binding = DataBindingUtil.getBinding<GankItemWelfareBinding>(holder.itemView)
        binding?.gankDayBean = item

//        瀑布流布局
//        int position = helper.getAdapterPosition();
//        if (position % 2 == 0) {
//            ivPic.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, DensityUtil.dp2px(250f)));
//        } else {
//            ivPic.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,  DensityUtil.dp2px(300f)));
//        }
    }
}